tensorflow/core/kernels/mkl/mkl_qmatmul_op.cc

```diff
@@ -633,89 +634,121 @@ class MklDnnQuantizedMatMulOp
 
       std::vector<dnnl::primitive> net;
       float out_scale;
-      auto scaled_bias_md = mkldnn_matmul_fwd_pd->bias_desc();
-      TensorShape scaled_bias_shape;
-      scaled_bias_shape.AddDim(
-          (scaled_bias_md.get_size() / sizeof(TSCALED_BIAS)));
-      OP_REQUIRES_OK(context, context->allocate_temp(
-                                  DataTypeToEnum<TSCALED_BIAS>::v(),
-                                  scaled_bias_shape, temp_scaled_bias_tensor));
-      void* scaled_bias_buf = static_cast<void*>(
-          temp_scaled_bias_tensor->flat<TSCALED_BIAS>().data());
-      // If the bias is float and input quantize is MIN_FIRST, bias has to be
-      // compensated with B's32 = Q'a * Qw * Bf32 + Q'a * Qw * Min(Af32) * 1 *
-      // Wf32.
-      if (mode_ == QUANTIZE_MODE_MIN_FIRST) {
-        int k = weight_tensor.dim_size(0);
-        int n = weight_tensor.dim_size(1);
-        TSCALED_BIAS* comp_bias = (TSCALED_BIAS*)scaled_bias_buf;
-
-        qint8* wt_buf = static_cast<qint8*>(
-            const_cast<qint8*>(weight_tensor.flat<qint8>().data()));
 
-        const Tbias* bias_buf = static_cast<Tbias*>(
-            const_cast<Tbias*>(bias_tensor.flat<Tbias>().data()));
-
-        float qa_amin = 255 * min_input / (max_input - min_input);
-
-        out_scale = (255.0 * 127.0) /
-                    ((max_input - min_input) *
-                     std::max(std::abs(max_weight), std::abs(min_weight)));
-        for (int j = 0; j < n; ++j) {
-          int x = 0;
-          for (int i = 0; i < k; ++i) {
-            x += wt_buf[i * n + j];
-          }
-          if (std::is_same<Tbias, qint32>::value) {
-            // Starting with oneDNN v3.0, bias is expected to be dequantized to
-            // float32.
-            comp_bias[j] = static_cast<float>(bias_buf[j]) / out_scale;
-          } else {
-            // Bias is float32 but still needs to be compensated.
-            comp_bias[j] = static_cast<float>(bias_buf[j]) + (x * qa_amin);
+      bool is_cached_bias_valid = false;
+      bool is_bias_cache_empty = this->IsBiasCacheEmpty();
+      if (!is_bias_cache_empty) {
```

Comment:
I did not change this since `is_bias_cache_empty` is used in two places in the code. Please let me know if you still think calling the method is better.

Reply:
Let's keep `is_bias_cache_empty`. I saw the second place later and forgot to remove my comment about it. Thank you!